photon-api/src/main/scala/com/linkedin/photon/ml/evaluation/RMSEEvaluator.scala

```diff
@@ -31,7 +31,9 @@ object RMSEEvaluator extends SingleEvaluator {
    */
   override def evaluate(scoresAndLabelsAndWeights: RDD[(Double, Double, Double)]): Double = {
 
-    val squaredLoss = SquaredLossEvaluator.evaluate(scoresAndLabelsAndWeights)
+    // squared loss is calculated as 1/2 (z - y)^2 for mathematical convenience
+    // however, RMSE should be using (z - y)^2 for calculation, hence the multiplying by 2 here:
+    val squaredLoss = 2 * SquaredLossEvaluator.evaluate(scoresAndLabelsAndWeights)
```

Comment:
I think the better place for this multiplication is in the `SquaredLossEvaluator`, as that is also a validation evaluator using the training code for convenience. It's also possible that if someone were to enable both the `SquaredLossEvaluator` and the `RMSEEvaluator`, they could manually compare the two values (since one is computed from the other) and see that they don't match.

Reply:
makes sense, updated.